src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -238,14 +232,16 @@ protected override void SetImplementationFlagsCore(MethodImplAttributes attribut
                 {
                     ArgumentNullException.ThrowIfNull(_parameterTypes[i] = parameterTypes[i], nameof(parameterTypes));
                 }
+
+                _parameterTypeOptionalCustomModifiers = parameterTypeOptionalCustomModifiers;
+                _parameterTypeRequiredCustomModifiers = parameterTypeRequiredCustomModifiers;
             }
-            // TODO: Add support for other parameters: returnTypeRequiredCustomModifiers, returnTypeOptionalCustomModifiers, parameterTypeRequiredCustomModifiers and parameterTypeOptionalCustomModifiers
         }
 
         public override string Name => _name;
         public override MethodAttributes Attributes => _attributes;
         public override CallingConventions CallingConvention => _callingConventions;
-        public override TypeBuilder DeclaringType => _declaringType;
```

Comment:
Ideas on why this was TypeBuilder previously?

Reply:
It was referenced internally to set the `TypeBuilder` of a generic method definition, to avoid casting I assume